app/policies/ticket_desk_policy.rb

```diff
@@ -0,0 +1,22 @@
+class TicketDeskPolicy < ApplicationPolicy
+  class Scope
+    attr_reader :user, :scope
+
+    def initialize(user, scope)
+      @user = user
+      @scope = scope
+    end
+
+    def resolve
+      if user.admin?
+        scope.all
+      else
+        scope.where(published: true)
+      end
+    end
+
+    def update?
+      resource.user == user
```

Comment:
In some places you are using `user`, but in some places `user_id`. Could you try to make it consistent? Which one way would you prefer?